src/clients/python/experimental_api_v2/examples/simple_grpc_v2_sequence_infer_client.py

```diff
@@ -0,0 +1,192 @@
+        value_data = np.full(shape=[batch_size, 1],
+                             fill_value=value,
+                             dtype=np.int32)
+        inputs = []
+        inputs.append(grpcclient.InferInput('INPUT'))
+        # Initialize the data
+        inputs[0].set_data_from_numpy(value_data)
+        outputs = []
+        outputs.append(grpcclient.InferOutput('OUTPUT'))
+        sequence_metadata.add_request(inputs,
+                                      outputs,
+                                      is_sequence_end=(count == len(values)))
+        count = count + 1
+
+    # Issue the asynchronous sequence inference.
+    triton_client.async_sequence_infer(response_pool=response_pool,
+                                       sequence_metadata=sequence_metadata,
+                                       model_name=model_name,
+                                       model_version=model_version)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v',
+                        '--verbose',
+                        action="store_true",
+                        required=False,
+                        default=False,
+                        help='Enable verbose output')
+    parser.add_argument(
+        '-u',
+        '--url',
+        type=str,
+        required=False,
+        default='localhost:8001',
+        help='Inference server URL and it gRPC port. Default is localhost:8001.'
+    )
+    parser.add_argument('-d',
+                        '--dyna',
+                        action="store_true",
+                        required=False,
+                        default=False,
+                        help='Assume dynamic sequence model')
+    parser.add_argument('-o',
+                        '--offset',
+                        type=int,
+                        required=False,
+                        default=0,
+                        help='Add offset to sequence ID used')
+
+    FLAGS = parser.parse_args()
+
+    try:
+        triton_client = grpcclient.InferenceServerClient(FLAGS.url)
+    except Exception as e:
+        print("context creation failed: " + str(e))
+        sys.exit()
+
+    # We use the custom "sequence" model which takes 1 input
+    # value. The output is the accumulated value of the inputs. See
+    # src/custom/sequence.
+    model_name = "simple_sequence"
+    model_version = ""
+    batch_size = 1
+
+    values = [11, 7, 5, 3, 2, 0, 1]
+
+    # Will use two sequences and send them asynchronously. Note the
+    # sequence IDs should be non-zero because zero is reserved for
+    # non-sequence requests.
+    sequence_id0 = 1000 + FLAGS.offset * 2
+    sequence_id1 = 1001 + FLAGS.offset * 2
+
+    result0_list = []
+    result1_list = []
+
+    user_data_0 = UserData()
+    user_data_1 = UserData()
+
+    # Create the thread pool with the size as expected
+    # number of concurrent sequences.
+    response_pool = get_stream_response_processor_pool(2)
```

Comment:
So if I want to have 1000 simultaneous sequences I need to have 1000 GRPC threads? Is that because GRPC requires a dedicated thread for each stream?

Reply:
I'd like to understand this more and how it compares to what our current GRPC client does. Seems like perhaps this pool should be associated with a sequence then and should instead be a "global" in the InferenceServerClient. I will set up a meeting tomorrow so we can discuss.